hugr/src/hugr/ident.rs

```diff
@@ -6,8 +6,12 @@ use regex::Regex;
 use smol_str::SmolStr;
 use thiserror::Error;
 
+pub static PATH_COMPONENT_REGEX_STR: &str = r"[\w--\d]\w*";
```

Comment:
```suggestion
pub static PATH_COMPONENT_REGEX_STR: &str = r"[\w_\d]\w*";
```
Not sure if we allow `-`s anywhere, but I'd be surprised if we wanted path components to start with `-`.

Hmmm. We probably *do* allow such, though! So maybe just `-_`